src/pokeblock_feed.c

```diff
@@ -0,0 +1,1012 @@
+{
+    FreeOamMatrix(gSprites[spriteID].oam.matrixNum);
+    gSprites[spriteID].oam.affineMode = 3;
+    if (!a1)
+        gSprites[spriteID].affineAnims = sSpriteAffineAnimTable_84121A4;
+    else
+        gSprites[spriteID].affineAnims = sSpriteAffineAnimTable_84121A8;
+    InitSpriteAffineAnim(&gSprites[spriteID]);
+}
+
+static u8 CreatePokeblockSprite(void)
+{
+    u8 spriteID = CreateSprite(&sThrownPokeblockSpriteTemplate, 174, 84, 1);
+    gSprites[spriteID].data0 = -12;
+    gSprites[spriteID].data1 = 1;
+    return spriteID;
+}
+
+static void SpriteCB_ThrownPokeblock(struct Sprite* sprite)
+{
+    sprite->pos1.x -= 4;
+    sprite->pos1.y += sprite->data0;
+    sprite->data0 += sprite->data1;
+    if (sprite->data0 == 10)
+        DestroySprite(sprite);
+}
+
+static void sub_81481DC(void)
+{
+    u8 animID, i;
+
+    gUnknown_03005F34 = 1;
+    animID = sNatureToMonPokeblockAnim[gPokeblockFeedMonNature][0];
+    for (i = 0; i < 8; i++, animID++)
+    {
+        gUnknown_03005F34 += sMonPokeblockAnims[animID][4];
+        if (sMonPokeblockAnims[animID][9] == 1)
+            break;
+    }
+}
+
+static void sub_814825C(void)
+{
+    switch (gUnknown_03005F3C)
+    {
+    case 0:
+        gUnknown_03005F40 = sNatureToMonPokeblockAnim[gPokeblockFeedMonNature][0];
+        gPokeblockFeedPokeSprite = &gSprites[gPokeblockFeedMonSpriteID];
+        gPokeblockFeedPokeSpriteCopy = *gPokeblockFeedPokeSprite;
+        gUnknown_03005F3C = 10;
+        break;
+    case 1 ... 9:
+        break;
+    case 10:
+        sub_8148540();
+        if (sNatureToMonPokeblockAnim[gPokeblockFeedMonNature][1] != 0)
+        {
+            gPokeblockFeedPokeSprite->oam.affineMode = 3;
+            gPokeblockFeedPokeSprite->oam.matrixNum = 0;
+            gPokeblockFeedPokeSprite->affineAnims = sSpriteAffineAnimTable_8412050;
+            InitSpriteAffineAnim(gPokeblockFeedPokeSprite);
+        }
+        gUnknown_03005F3C = 50;
+    case 50:
+        if (sNatureToMonPokeblockAnim[gPokeblockFeedMonNature][1] != 0)
+        {
+            if (gPokeblockMonNotFlipped == 0)
+                StartSpriteAffineAnim(gPokeblockFeedPokeSprite, sNatureToMonPokeblockAnim[gPokeblockFeedMonNature][1] + 10);
+            else
+                StartSpriteAffineAnim(gPokeblockFeedPokeSprite, sNatureToMonPokeblockAnim[gPokeblockFeedMonNature][1]);
+        }
+        gUnknown_03005F3C = 60;
+        break;
+    case 60:
+        if (sub_81485CC() == 1)
+        {
+            if (gUnknown_03005FA0[9] == 0)
+            {
+                gUnknown_03005F40++;
+                sub_8148540();
+                gUnknown_03005F3C = 60;
+            }
+            else
+            {
+                FreeOamMatrix(gPokeblockFeedPokeSprite->oam.matrixNum);
+                gUnknown_03005F3C = 70;
+            }
+        }
+        break;
+    case 70:
+        FreePokeSpriteMatrix();
+        gUnknown_03005F40 = 0;
+        gUnknown_03005F3C = 0;
+        break;
+    case 71 ... 90:
```

Comment:
Maybe try changing this to `case 90` here and getting rid of the `case 1 ... 9` above. I've encountered things like this in other files, so those intermediate cases are just skipping the switch.